### PR TITLE
Add section about user impact to PR template

### PR DESCRIPTION
**Pull request by sherifnada**
## What
Add a section to the PR template which prompts the developer to think about the user impact of their change

the goal of this is to get the developer to think about the end impact this particular PR they are raising will have on the user especially if it contains a breaking change. If there is a breaking change, then we want to make sure it is:

1. very thoroughly considered
2. is communicated to users clearly perhaps via something like airbytehq/airbyte-internal-issues#331

this way, if there are any breaking issues or action required from the user, or just anything they really need to be aware of, this section will bring that to the forefront

### CODE REVIEW

File: .github/pull_request_template.md [hunk 1]
```diff
@@ -9,12 +9,16 @@
...
+Are there any breaking changes? 
+What is the end result perceived by the user?
+
 ## Pre-merge Checklist
 Expand the relevant checklist and delete the others. 
 
 <details><summary> <strong> New Connector </strong></summary>
 <p>
-   
+
```
Comment (tuliren) [inline .github/pull_request_template.md:21]:
```suggestion
```

File: .github/pull_request_template.md [hunk 2]
```diff
@@ -9,12 +9,16 @@
 1. `x.java`
 2. `y.python`
 
+## 🚨 User Impact 🚨
+Are there any breaking changes? 
+What is the end result perceived by the user?
```
Comment (tuliren) [inline .github/pull_request_template.md:14]: I think it's probably more useful to put this in the connector README than in PR description? People usually won't check this.

**Review by tuliren — APPROVED** [review]

File: .github/pull_request_template.md (hunk 2 above)
Comment (sherifnada) [reply .github/pull_request_template.md:14]: the goal of this is to get the developer to think about the end impact this particular PR they are raising will have on the user especially if it contains a breaking change. If there is a breaking change, then we want to make sure it is: 
1) very thoroughly considered
2) is communicated to users clearly perhaps via something like #8004 

this way, if there are any breaking issues or action required from the user, or just anything they really need to be aware of, this section will bring that to the forefront.

Comment (tuliren) [reply .github/pull_request_template.md:14]: Got it. Make sense.

I was actually thinking about the connector doc, but I wrote down "README" by mistake. It is probably also worth adding alert about breaking changes to the `CHANGELOG` section of the connector doc.

File: .github/pull_request_template.md [hunk 3]
```diff
@@ -9,12 +9,15 @@
...
+
 ## Pre-merge Checklist
 Expand the relevant checklist and delete the others. 
 
 <details><summary> <strong> New Connector </strong></summary>
 <p>
-
```
Comment (tuliren) [inline .github/pull_request_template.md:17]: Oh, sorry that I did not mean to delete this whole line. Just wanted to remove the newly-introduced spaces.

File: .github/pull_request_template.md [hunk 4]
```diff
@@ -9,12 +9,15 @@
...
 
+## 🚨 User Impact 🚨
+Are there any breaking changes? If yes, please make sure to include it here and in any changelogs with the 🚨🚨 emoji
+What is the end result perceived by the user?
+
 ## Pre-merge Checklist
 Expand the relevant checklist and delete the others. 
 
 <details><summary> <strong> New Connector </strong></summary>
 <p>
```
Comment (sherifnada) [inline .github/pull_request_template.md:20]:
```suggestion
<p>

```

**Pull request merged**